### ISSUE TITLE
GSAP & AMD:  requiring both TweenMax and TimelineMax

### ISSUE THREAD

**Issue by theprojectsomething** [janpaepke/ScrollMagic#427]
I believe TimelineMax is included in the TweenMax library, yet ScrollMagic requires each file separately in it's AMD setup (adding 6kb to the build).

**Comment by homerjam** [comment]
> I believe TimelineMax is included in the TweenMax library

This is correct

**Comment by VarinderS** [comment]
Yes, even though it needs each file separately, it isn't required as TweenMax loads TimelineMax anyway.

So, I went ahead and removed TimelineMax dependency from ```define(['ScrollMagic', 'TweenMax', 'TimelineMax'], factory);``` [here](https://github.com/janpaepke/ScrollMagic/blob/master/dev/src/plugins/animation.gsap.js#L22) changing it to ```define(['ScrollMagic', 'TweenMax'], factory);```

Builds fine for now

**Comment by hellorich** [comment]
I hit this issue as well. Is this due to a change in Greensock that the plugin hasn't kept up with, or something else I'm misunderstanding?

**Comment by massic80** [comment]
I figure out they are left separately because one could just use TimelineMax (with TweenLite?) and not TweenMax.

**Comment by janpaepke** [comment]
Over ten years — sorry for the wait.

This was a v2 AMD/module configuration issue. v3 is a proper ES module with native TypeScript and zero dependencies — no GSAP plugin bundling, no AMD shim configuration. You import ScrollMagic directly and pair it with whatever animation library you prefer (GSAP, Motion, anime.js, etc.) through the event API. No special integration files needed.

---
**ScrollMagic v3 is now available** — a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` — [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
